doc/DataPlaneCodeGeneration/DeveloperDrivenEvolution.md

```diff
@@ -97,7 +97,28 @@ public virtual async Task<Response<MetricFeedback>> GetMetricFeedbackValuesAsync
 If the suggested rename does not apply to your scenario, you could pick the best name fitting your API.
 
 ## Implement the method
-You could call the protocol method inside your convenience method and map the returned raw response to model. See different scenarios and corresponding examples below. 
+You could call the protocol method inside your convenience method and map the returned raw response to model.
+
+In the implementation, you should only have one `DiagnosticScope` inside instead of creating `DiagnosticScope` for both protocol method and convenience method. Reasons are:
+- We want to reduce the telemetry costs.
+- Trace should correspond to logical operations and not implementations
+
+To achieve this, our guidance is:
+- Don't open a scope in the grow-up method if it is just a simple wrapper (i.e. if the only thing in the convenience API is serialization and deserialization and the protocol method and convenience API have the same name). Logging will help developers debug any exceptions thrown from the convenience method.
+- If the grow-up method diverges much from the protocol method (i.e. if they have different names or code in the convenience method could take much time), do open a scope in the convenience API and turn on suppression of inner scopes.
+- If you want to add attributes or otherwise modify scopes, do that in the convenience method.
+
+To turn on the suppression, pass `ture` to `ClientDiagnostics` in the client constructor.
+```C#
+public MetricsAdvisorClient(Uri endpoint, MetricsAdvisorKeyCredential credential, MetricsAdvisorClientsOptions options)
+{
+    ...
+    ClientDiagnostics = new ClientDiagnostics(options, true);
```

Comment:
Don't quite understand about this.  As MetricsAdvisorClient is generated, do you mean we need to modify the generated code?

Reply:
It is generated. Updated.

Reply:
If the code is generated, we probably don't need to show it in this document, since this document is primarily focused on convenience methods.  I think mentioning that this is the default without showing the code is sufficient.